ExtremeValueProject/WeakConvergenceCDF.lean

```diff
@@ -38,15 +38,61 @@ lemma forall_exists_subdivision_dist_apply_lt_of_dense_of_continuous {D : Set 
       (cs 0 = a) ∧ (cs (Fin.last _) = b) ∧ (Monotone cs) ∧ (∀ k, cs k ∈ D) ∧
       (∀ (j : Fin k), ∀ x ∈ Icc (cs j) (cs j.succ), ∀ y ∈ Icc (cs j) (cs j.succ),
         dist (f x) (f y) < ε) := by
-  sorry -- **Issue #17**
+  let I : Set ℝ := Icc a b
+  have hI_compact : IsCompact I := isCompact_Icc
+  have hI_nonempty : I.Nonempty := nonempty_Icc.mpr (le_of_lt a_lt_b)
+  have hf_cont_I : ContinuousOn f I := by exact Continuous.continuousOn f_cont
+  have hf_unif_cont : UniformContinuousOn f I := hI_compact.uniformContinuousOn_of_continuous hf_cont_I
+  have h_δ : ∃ δ > 0, ∀ x ∈ I, ∀ y ∈ I, dist x y < δ → dist (f x) (f y) < ε := by
+    rw [Metric.uniformContinuousOn_iff] at hf_unif_cont
+    specialize hf_unif_cont ε
+    apply hf_unif_cont
+    exact ε_pos
+  obtain ⟨δ, hδ_pos, hδ⟩ := h_δ
+  obtain ⟨k, cs, h_cs_0, h_cs_last, h_cs_mono, h_cs_D, h_cs_diff⟩ :=
+    forall_exists_subdivision_diff_lt_of_dense D_dense ha hb a_lt_b hδ_pos
+  have h_cs_bound : ∀ i : Fin k, ∀ x ∈ Icc (cs i) (cs i.succ), ∀ y ∈ Icc (cs i) (cs i.succ), dist (f x) (f y) < ε := by
+    intro i x hx y hy
+    have hx_I : x ∈ I := by
+      unfold I
+      have h_lower : a ≤ cs i := by
+        rw [← h_cs_0]
+        apply h_cs_mono
+        apply Fin.zero_le
+      have h_upper : cs i.succ ≤ b := by
+        rw [← h_cs_last]
+        apply h_cs_mono
+        exact Fin.le_last i.succ
```

Comment:
Combining steps to a shorter proof here can in my opinion be done here without sacrificing clarity.

```suggestion
      have h_lower : a ≤ cs i := by simpa [← h_cs_0] using h_cs_mono (Fin.zero_le _)
      have h_upper : cs i.succ ≤ b := by simpa [← h_cs_last] using h_cs_mono (Fin.le_last i.succ)
```